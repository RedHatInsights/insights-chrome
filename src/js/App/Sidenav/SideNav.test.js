import React from 'react';
import ConnectedSideNav from './SideNav';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';

describe('ConnectedSideNav', () => {
  let initialState;
  let mockStore;

  beforeEach(() => {
    mockStore = configureStore();
    initialState = {
      chrome: {
        activeTechnology: 'someTechnology',
        activeLocation: 'someLocation',
      },
    };
  });
  it('should render correctly', () => {
    const store = mockStore(initialState);
    const { container } = render(
      <Provider store={store}>
        <ConnectedSideNav />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
  it('should render correctly part 2', () => {
    const store = mockStore(initialState);
    const { container } = render(
      <Provider store={store}>
        <ConnectedSideNav />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
