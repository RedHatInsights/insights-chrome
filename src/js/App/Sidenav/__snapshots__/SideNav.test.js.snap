// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedSideNav should render correctly 1`] = `
<div>
  <section
    class="ins-c-app-switcher--loading"
  >
    <div
      class="ins-c-skeleton ins-c-skeleton__lg ins-m-dark"
    >
       
    </div>
  </section>
  <nav
    aria-label="Insights Global Navigation"
    class="pf-c-nav"
    data-ouia-component-id="OUIA-Generated-Nav-1"
    data-ouia-component-type="PF4/Nav"
    data-ouia-safe="false"
  >
    <ul
      class="pf-c-nav__list"
    >
      <li
        class="pf-c-nav__item"
        data-ouia-component-id="OUIA-Generated-NavItem-1"
        data-ouia-component-type="PF4/NavItem"
        data-ouia-safe="true"
      >
        <a
          class="pf-c-nav__link"
          href="http://localhost//undefined"
        >
          <div
            class="ins-c-skeleton ins-c-skeleton__lg ins-m-dark ins-c-skeleton__link"
          >
             
          </div>
           
        </a>
      </li>
      <li
        class="pf-c-nav__item"
        data-ouia-component-id="OUIA-Generated-NavItem-2"
        data-ouia-component-type="PF4/NavItem"
        data-ouia-safe="true"
      >
        <a
          class="pf-c-nav__link"
          href="http://localhost//undefined"
        >
          <div
            class="ins-c-skeleton ins-c-skeleton__lg ins-m-dark ins-c-skeleton__link"
          >
             
          </div>
           
        </a>
      </li>
      <li
        class="pf-c-nav__item"
        data-ouia-component-id="OUIA-Generated-NavItem-3"
        data-ouia-component-type="PF4/NavItem"
        data-ouia-safe="true"
      >
        <a
          class="pf-c-nav__link"
          href="http://localhost//undefined"
        >
          <div
            class="ins-c-skeleton ins-c-skeleton__lg ins-m-dark ins-c-skeleton__link"
          >
             
          </div>
           
        </a>
      </li>
      <li
        class="pf-c-nav__item"
        data-ouia-component-id="OUIA-Generated-NavItem-4"
        data-ouia-component-type="PF4/NavItem"
        data-ouia-safe="true"
      >
        <a
          class="pf-c-nav__link"
          href="http://localhost//undefined"
        >
          <div
            class="ins-c-skeleton ins-c-skeleton__lg ins-m-dark ins-c-skeleton__link"
          >
             
          </div>
           
        </a>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`ConnectedSideNav should render correctly part 2 1`] = `
<div>
  <section
    class="ins-c-app-switcher--loading"
  >
    <div
      class="ins-c-skeleton ins-c-skeleton__lg ins-m-dark"
    >
       
    </div>
  </section>
  <nav
    aria-label="Insights Global Navigation"
    class="pf-c-nav"
    data-ouia-component-id="OUIA-Generated-Nav-2"
    data-ouia-component-type="PF4/Nav"
    data-ouia-safe="false"
  >
    <ul
      class="pf-c-nav__list"
    >
      <li
        class="pf-c-nav__item"
        data-ouia-component-id="OUIA-Generated-NavItem-5"
        data-ouia-component-type="PF4/NavItem"
        data-ouia-safe="true"
      >
        <a
          class="pf-c-nav__link"
          href="http://localhost//undefined"
        >
          <div
            class="ins-c-skeleton ins-c-skeleton__lg ins-m-dark ins-c-skeleton__link"
          >
             
          </div>
           
        </a>
      </li>
      <li
        class="pf-c-nav__item"
        data-ouia-component-id="OUIA-Generated-NavItem-6"
        data-ouia-component-type="PF4/NavItem"
        data-ouia-safe="true"
      >
        <a
          class="pf-c-nav__link"
          href="http://localhost//undefined"
        >
          <div
            class="ins-c-skeleton ins-c-skeleton__lg ins-m-dark ins-c-skeleton__link"
          >
             
          </div>
           
        </a>
      </li>
      <li
        class="pf-c-nav__item"
        data-ouia-component-id="OUIA-Generated-NavItem-7"
        data-ouia-component-type="PF4/NavItem"
        data-ouia-safe="true"
      >
        <a
          class="pf-c-nav__link"
          href="http://localhost//undefined"
        >
          <div
            class="ins-c-skeleton ins-c-skeleton__lg ins-m-dark ins-c-skeleton__link"
          >
             
          </div>
           
        </a>
      </li>
      <li
        class="pf-c-nav__item"
        data-ouia-component-id="OUIA-Generated-NavItem-8"
        data-ouia-component-type="PF4/NavItem"
        data-ouia-safe="true"
      >
        <a
          class="pf-c-nav__link"
          href="http://localhost//undefined"
        >
          <div
            class="ins-c-skeleton ins-c-skeleton__lg ins-m-dark ins-c-skeleton__link"
          >
             
          </div>
           
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
